ansible/cluster.yml

```diff
@@ -25,10 +25,51 @@
     - nodes
   sudo: yes
   roles:
-    - flannel
+    - { role: flannel, when: networking == 'flannel' }
   tags:
     - flannel
 
+# install opencontrail
+- hosts:
+    - masters[0]
+  sudo: yes
+  roles:
+    - { role: opencontrail-kmod, when: networking == 'opencontrail' }
+  tags:
+    - opencontrail-kmod
+
+- hosts:
+    - masters
+  gather_facts: yes
+  sudo: yes
+  tags:
+    - opencontrail
+    - opencontrail-gateway
```

Comment:
I'm just asking, can you do it in 2 top level roles?  Then do the kmod/gateway/controller/node stuff inside that one role?  In teh second top level role you'd have to do the master/node provision.

Two large roles might be horrible for you, if so, this is ok. I know this is the "ansible" way but I'd rather hide opencontrail for non-contrail users....